libsolidity/ASTJsonConverter.cpp

```diff
@@ -0,0 +1,486 @@
+/*
+    This file is part of cpp-ethereum.
+
+    cpp-ethereum is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    cpp-ethereum is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @author Lefteris <lefteris@ethdev.com>
+ * @date 2015
+ * Converts the AST into json format
+ */
+
+#include <libsolidity/ASTJsonConverter.h>
+#include <libsolidity/AST.h>
+
+using namespace std;
+
+namespace dev
+{
+namespace solidity
+{
+
+void ASTJsonConverter::addKeyValue(Json::Value& _obj, string const& _key, string const& _val)
+{
+	// special handling for booleans
+	if (_key == "const" || _key == "public" || _key == "local" ||
+		_key == "lvalue" || _key == "local_lvalue" || _key == "prefix")
+		_obj[_key] = (_val == "1") ? true : false;
+	else
+		// else simply add it as a string
+		_obj[_key] = _val;
+}
+
+void ASTJsonConverter::addJsonNode(string const& _nodeName,
+								   initializer_list<pair<string const, string const>> _list,
+								   bool _hasChildren = false)
+{
+	Json::Value node;
+
+	node["name"] = _nodeName;
+	if (_list.size() !=0)
+	{
+		Json::Value attrs;
+		for (auto &e: _list)
+			addKeyValue(attrs, e.first, e.second);
+		node["attributes"] = attrs;
+	}
+
+	(*m_jsonNodePtrs.top()).append(node);
```

Comment:
This is the main reason why I spend a hell of a lot of time debugging this code. There is an internal jsoncpp assertion that checks for the type of the Json::Value and triggers an internal library assert if it's a pointer. I have no idea why.